pkg/schemadsl/generator/generator.go

```diff
@@ -16,6 +25,43 @@ const Ellipsis = "..."
 // MaxSingleLineCommentLength sets the maximum length for a comment to made single line.
 const MaxSingleLineCommentLength = 70 // 80 - the comment parts and some padding
 
+// GenerateSchema generates a DSL view of the given schema.
+func GenerateSchema(definitions []compiler.SchemaDefinition) (string, bool) {
+	generated := make([]string, 0, len(definitions))
+	result := true
+	for _, definition := range definitions {
+		switch def := definition.(type) {
+		case *core.CaveatDefinition:
+			generatedCaveat, ok := GenerateCaveatSource(def)
+			result = result && ok
+			generated = append(generated, generatedCaveat)
+
+		case *core.NamespaceDefinition:
+			generatedSchema, ok := GenerateSource(def)
+			result = result && ok
+			generated = append(generated, generatedSchema)
+
+		default:
+			panic(fmt.Sprintf("unknown type of definition %T in GenerateSchema", def))
```

Comment:
why not returning an error instead of panic? the `runOperation` method from `wasm` seems to return an error, so you could bubble an error up instead of panic

that could be done in the other methods of the generator too

Reply:
My personal take is that, regardless whether this is used in the actual service or in debug tooling, we should use `panic` as it is intended: for an unrecoverable error that compromises the integrity of the process running. If we err on the side of panicking instead of returning an error, we may end up taking a spicedb instance down for something it could have handled gracefully.